File: app/src/main/java/com/botijasoftware/utils/ArrayBufferFloat.java
package com.botijasoftware.utils;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.nio.FloatBuffer;

public class ArrayBufferFloat extends ArrayBuffer {
	private FloatBuffer mBuffer;
	private final static int float_size = Float.SIZE/8;

	public ArrayBufferFloat(VertexBufferDefinition vbdefinition, int maxelements) {
		super(vbdefinition,maxelements);

		mBuffer = ByteBuffer.allocateDirect(
				vbdefinition.getDataSize() * vbdefinition.mSize * maxelements * float_size)
				.order(ByteOrder.nativeOrder()).asFloatBuffer();
	}
	
	public FloatBuffer getBuffer() {
		return mBuffer;
	}
	
	public void put(int index, float value) {
		mBuffer.put(index, value);
	}
	
	public void put(float[] data, int offset) {
		mBuffer.put(data, offset, data.length - offset);
		mBuffer.position(0);
	}
	
	public void put(float[] data) {
		mBuffer.put(data, 0, data.length);
		mBuffer.position(0);
	}
	
	public void put(byte value) {
		mBuffer.put(value);
	}
	
	public void put(short value) {
		mBuffer.put(value);
	}
	
	public void put(int value) {
		mBuffer.put(value);
	}
	
	public void put(float value) {
		mBuffer.put(value);
	}
	
	public void position(int i) {
		mBuffer.position(i);
	}
	
	public int position() {
		return mBuffer.position();
	}

}

File: app/src/main/java/com/botijasoftware/utils/ArrayBufferFloat.kt
package com.botijasoftware.utils

import java.nio.ByteBuffer
import java.nio.ByteOrder
import java.nio.FloatBuffer

class ArrayBufferFloat(vbdefinition: VertexBufferDefinition, maxelements: Int) : ArrayBuffer(vbdefinition, maxelements) {
    override val buffer: FloatBuffer

    init {

        buffer = ByteBuffer.allocateDirect(
                vbdefinition.dataSize * vbdefinition.mSize * maxelements * float_size)
                .order(ByteOrder.nativeOrder()).asFloatBuffer()
    }

    override fun put(index: Int, value: Float) {
        buffer.put(index, value)
    }

    override fun put(data: FloatArray, offset: Int) {
        buffer.put(data, offset, data.size - offset)
        buffer.position(0)
    }

    override fun put(data: FloatArray) {
        buffer.put(data, 0, data.size)
        buffer.position(0)
    }

    override fun put(value: Byte) {
        buffer.put(value.toFloat())
    }

    override fun put(value: Short) {
        buffer.put(value.toFloat())
    }

    override fun put(value: Int) {
        buffer.put(value.toFloat())
    }

    override fun put(value: Float) {
        buffer.put(value)
    }

    override fun position(i: Int) {
        buffer.position(i)
    }

    override fun position(): Int {
        return buffer.position()
    }

    companion object {
        private val float_size = java.lang.Float.SIZE / 8
    }

}